### PR TITLE
[WIP] fix: Change default values for Horizontal and Vertical content alignment

### LINKED ISSUE #4038: Default alignment does not match UWP
## Current behavior

The `Uno.UI.FeatureConfiguration.Control.UseLegacyContentAlignment` is set to `false` and based on the `DependencyPropertyMixins.tt` the default `HorizontalContentAlignment`, `VerticalContentAlignment` is set to `Center`.

```
.Class("Control")
   .Property("HorizontalContentAlignment", "HorizontalAlignment", 
      "(Uno.UI.FeatureConfiguration.Control.UseLegacyContentAlignment ? 
       HorizontalAlignment.Left : HorizontalAlignment.Center)")
   .Property("VerticalContentAlignment", "VerticalAlignment", 
      "(Uno.UI.FeatureConfiguration.Control.UseLegacyContentAlignment ? 
       VerticalAlignment.Top : VerticalAlignment.Center)")			
```

## Expected behavior

UWP's default `HorizontalContentAlignment` is `Left` ([docs](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.control.horizontalcontentalignment?view=winrt-19041)) and `VerticalContentAlignment` is `Top` ([docs](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.control.verticalcontentalignment?view=winrt-19041))

Seems like the "legacy" behavior is actually the correct behavior 🤔

## How to reproduce it (as minimally and precisely as possible)

N/A

## Workaround

Explicitly set the `HorizontalContentAlignment` and `VerticalContentAlignment`.

## Environment

Nuget Package:

- [x] Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia
- [ ] Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia
- [ ] Uno.SourceGenerationTasks
- [ ] Uno.UI.RemoteControl / Uno.WinUI.RemoteControl
- [ ] Other: 

Nuget Package Version(s):

Affected platform(s):

- [x] iOS
- [x] Android
- [x] WebAssembly
- [x] WebAssembly renderers for Xamarin.Forms
- [x] macOS
- [x] Skia
  - [x] WPF
  - [x] GTK (Linux)
  - [x] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

IDE:

- [ ] Visual Studio 2017 (version: )
- [ ] Visual Studio 2019 (version: )
- [ ] Visual Studio for Mac (version: )
- [ ] Rider Windows (version: )
- [ ] Rider macOS (version: )
- [ ] Visual Studio Code (version: )

Relevant plugins:

- [ ] Resharper (version: )

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): fixes #4038

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

Default alignment is `Center`.

## What is the new behavior?

Default alignment matches UWP.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Quite likely a breaking change, but aligns with UWP and existing apps probably had to use workaround (explicitly aligning).

### CODE REVIEW

**Comment by jeromelaban** [comment]
We'll need to take a look at the screenshot tests to validate the behavior of this change.

File: src/Uno.UI/Mixins/DependencyPropertyMixins.tt [hunk 1]
```diff
@@ -6,8 +6,8 @@
 		.Namespace("Windows.UI.Xaml.Controls")
 
 			.Class("Control")
-				.Property("HorizontalContentAlignment", "HorizontalAlignment", "(Uno.UI.FeatureConfiguration.Control.UseLegacyContentAlignment ? HorizontalAlignment.Left : HorizontalAlignment.Center)")
-				.Property("VerticalContentAlignment", "VerticalAlignment", "(Uno.UI.FeatureConfiguration.Control.UseLegacyContentAlignment ? VerticalAlignment.Top : VerticalAlignment.Center)")				
+				.Property("HorizontalContentAlignment", "HorizontalAlignment", "HorizontalAlignment.Left")
```
Comment (dr1rrb) [inline src/Uno.UI/Mixins/DependencyPropertyMixins.tt:9]: Apparently this flag (`Uno.UI.FeatureConfiguration.Control.UseLegacyContentAlignment`) is used only here, you should either (by order of personal preferences):
* document it as useless / deprecated
* raise an exception if set to `true`
* flag it as deprecated
* remove it (breaking change alert!)

**Comment by MartinZikmund** [comment]
@davidjohnoliver So this might be a bit trickier, from [docs](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.control.verticalcontentalignment?view=winrt-19041):

> Also, the apparent "default value" of each individual control used in UI can be different. For example, a **Button control starts with the value Center.**

This might mean we would have to add a custom `SetValue` with `default` precedence to the constructors of the controls which use the "non-conventional" defaults 🤔

**Comment by MartinZikmund** [comment]
UWP Docs are actually wrong here. Seems the alignment should default to `Center`. Closing for now, if some discrepancy shows up, we can adjust specifically.

**Pull request closed without merging**